code-challenges/two-dimensional-arrays/challenges-12.test.js

```diff
@@ -0,0 +1,325 @@
+
+For this example, the total at 9:00 a.m. is 17 + 26 + 7 + 5 + 33, or 88 total cookies.
+
+Return the array of the total number of cookies sold per hour for all of the stores combined.
+------------------------------------------------------------------------------------------------ */
+
+const hoursOpen = ['9 a.m.', '10 a.m.', '11 a.m.', '12 p.m.', '1 p.m.', '2 p.m.', '3 p.m.', '4 p.m.', '5 p.m.', '6 p.m.', '7 p.m.', '8 p.m.'];
+
+const firstPike = [17, 18, 23, 24, 24, 12, 13, 27, 30, 20, 24, 18];
+const seaTac = [26, 5, 5, 59, 23, 39, 38, 20, 30, 7, 59, 43];
+const seattleCenter = [7, 14, 19, 22, 15, 4, 23, 27, 28, 23, 1, 29];
+const capHill = [5, 85, 58, 51, 50, 13, 33, 32, 47, 94, 31, 62];
+const alkiBeach = [33, 31, 147, 130, 27, 93, 38, 126, 141, 63, 46, 17];
+
+const cookieStores = [firstPike, seaTac, seattleCenter, capHill, alkiBeach];
+
+const grandTotal = (stores) => {
+  let result = [0];
+  for(let i = 0; i < hoursOpen.length; i++){
+    let total = 0;
+    for(let j = 0; j < stores.length; j++){
+      console.log(stores[j][i]);
+      total += stores[j][i];
+    }
+    result[i] = total;
+  }
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Pat has decided that he would also like to organize his data as objects containing the number of cookies sold per hour and the time.
+
+Here is sample data for the 9:00 sales: { sales: '88 cookies', time: '9 a.m.' }.
+
+Write a function named salesData that uses forEach to iterate over the hourlySales array and create an object for each hour. Return an array of the formatted data.
+------------------------------------------------------------------------------------------------ */
+
+const salesData = (hours, data) => {
+  let result = [];
+  hours.forEach((element, index) => {
+    console.log(index);
+    let object = {
+      sales: `${data[index]} cookies`,
+      time: element
+    };
+    result.push(object);
+  });
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named howManyTreats that will return the quantity of treats you need to pick up from the pet store today from this array.
+------------------------------------------------------------------------------------------------ */
+
+const errands = [
+  { store: 'Grocery store',
+    items: [ { name: 'Eggs', quantity: 12 }, { name: 'Milk', quantity: 1 }, { name: 'Apples', quantity: 3 }]
+  },
+  { store: 'Drug store',
+    items: [ { name: 'Toothpaste', quantity: 1 }, { name: 'Toothbrush', quantity: 3 }, { name: 'Mouthwash',quantity: 1 } ]
+  },
+  { store: 'Pet store',
+    items: [ { name: 'Cans of food', quantity: 8 }, { name: 'Treats', quantity: 24 }, { name: 'Leash', quantity: 1 } ]
+  }
+];
+
+const howManyTreats = (arr) => {
+  let quantity = 0;
+  arr.map(element => {
```

Comment:
Since you're not returning an array a `.forEach` might have been more appropriate